taskcluster/docker/linux/Dockerfile

```diff
@@ -149,10 +156,37 @@ RUN \
 
 ENV SHELL=/bin/bash \
     HOME=/builds/worker \
-    PATH=/builds/worker/.local/bin:$PATH
+    PATH=/builds/worker/.local/bin:/builds/worker/.cargo/bin:$PATH
 
 VOLUME /builds/worker/checkouts
 VOLUME /builds/worker/.cache
 
-# run-task expects to run as root
+# Switch to the worker user to install Rust
+# After startup, run-task will downgrade it's permissions to be this user
+USER worker
+
+# Rust setup
+ENV RUST_BACKTRACE=1 \
+    CARGO_INCREMENTAL=1
```

Comment:
rustup-setup.sh has CARGO_INCREMENTAL as 0 - is the change here intentional? It also sets `RUSTFLAGS='-Dwarnings'` in some cases - we probably still want that?

Reply:
The `CARGO_INCREMENTAL` change was unintentional.  I'll fix that and also combine this with the other `ENV` statement above, not sure why I made 2 separate ones.

I wasn't sure about the `-Dwarnings`.  `rustup-setup.sh` was disabling them for releases do we want to keep that?  It feels a bit risky to me to release code that doesn't normally pass CI.  Could we disable the warning from the rust code instead?